data-access-layer/bosh/BoshDirectorClient.js

```diff
@@ -33,9 +36,56 @@ class BoshDirectorClient extends HttpClient {
     this.deploymentIpsCache = {};
     this.cacheLoadInProgressForDeployment = {};
     this.cacheLoadInProgress = false;
+    this.uaaObjects = {};
+    //populate UAA objects if uaa based auth is being used by any of the directors
+    this.determineDirectorAuthenticationMethod();
     this.ready = this.populateConfigCache();
   }
 
+  determineDirectorAuthenticationMethod() {
+    _.forEach(config.directors, (directorConfig) => {
+      if (_.get(directorConfig, 'uaa.uaa_auth', undefined)) {
```

Comment:
Can you be more specific in the `if` condition here? I am guessing you're only checking for existence of the value?